syntax = "proto3";

package com.baidu.acu.pie;

message InitRequest {
    bool enable_long_speech = 1;
    bool enable_chunk = 2;
    bool enable_flush_data = 3;

    string product_id = 4;
    uint32 sample_point_bytes = 5;
    double send_per_seconds = 6;
    double sleep_ratio = 7;

    string app_name = 8;
    uint32 log_level = 9;
}

message AudioFragmentRequest {
    bytes audio_data = 1;
}

message AudioFragmentResponse {
    int32 error_code = 1;
    string error_message = 2;
    string start_time = 3;
    string end_time = 4;
    string result = 5;
    bool completed = 6;
    string serial_num = 7;
}

service AsrService {
    rpc send (stream AudioFragmentRequest) returns (stream AudioFragmentResponse);
}